packages/app/components/settings/tabs/billing.tsx

```diff
@@ -105,6 +105,12 @@ const HistoryItem = memo(function HistoryItem({
 }: {
   item: PaymentsHistory;
 }) {
+  const handleViewPress = useCallback(() => {
+    for (let link of item.receipts) {
+      Linking.openURL(link);
+    }
```

Comment:
Why are you looping inside a press handler?

Reply:
good question. I copy-pasted it without even thinking 🤣

Reply:
cc - @alantoa

Reply:
receipts are an array of URLs because one payment intent may have resulted in multiple charges being made (including refunds most likely) and we get a receipt for each charge.
so we are probably wrong to use only one of them, now we are not sure multiple receipts arrays are not very common, so I think we can use this code first, at least it doesn't appear invisible bug, WDYT?

Reply:
But this would cause multiple openURL triggers. These will do what? Open InAppBrowsers? App Intents? For me this looks wrong. Only ok when these would be something like background fetches

Reply:
yeah, the issue is we don't have background fetches about this right now, and this page is only available on the web, we don't need to consider native.
now I came up with a plan that uses the Dropdown component to handle this, WDTY?

Reply:
I think I don't really understand this from just reading code.